storage/libradb/src/lib.rs

```diff
@@ -974,6 +984,37 @@ impl LibraDBTrait for LibraDB {
             .collect();
         Ok(events)
     }
+
+    fn get_state_proof(
+        &self,
+        known_version: u64,
+    ) -> Result<(
+        LedgerInfoWithSignatures,
+        ValidatorChangeProof,
+        AccumulatorConsistencyProof,
+    )> {
+        let ledger_info_with_sigs = self.ledger_store.get_latest_ledger_info()?;
+        let ledger_info = ledger_info_with_sigs.ledger_info();
+        let known_epoch = self.ledger_store.get_epoch(known_version)?;
+        let validator_change_proof = if known_epoch < ledger_info.epoch() {
+            let (ledger_infos_with_sigs, more) = self.get_epoch_change_ledger_infos(
+                known_epoch,
+                self.ledger_store.get_epoch(ledger_info.version())?,
+            )?;
+            ValidatorChangeProof::new(ledger_infos_with_sigs, more)
+        } else {
+            ValidatorChangeProof::new(vec![], /* more = */ false)
+        };
```

Comment:
maybe unnecessary, but should we handle the case where `known_epoch > ledger_info.epoch()` (e.g. throw error)?

Reply:
I just copy/pasted from original function. if known_epoch > ledger_info.epoch, proof will be just empty